app/src/main/java/at/bitfire/icsdroid/db/dao/CredentialsDao.kt

```diff
@@ -0,0 +1,59 @@
+package at.bitfire.icsdroid.db.dao
+
+import android.content.Context
+import android.database.SQLException
+import android.database.sqlite.SQLiteConstraintException
+import androidx.annotation.WorkerThread
+import androidx.room.Dao
+import androidx.room.Query
+import at.bitfire.icsdroid.db.AppDatabase
+import at.bitfire.icsdroid.db.entity.Credential
+import at.bitfire.icsdroid.db.entity.Subscription
+
+/**
+ * Creates an interface with all the credentials stored in the database.
+ * @see AppDatabase
+ * @see Credential
+ */
+@Dao
+interface CredentialsDao {
+    companion object {
+        /**
+         * Alias for [AppDatabase.getInstance] -> [AppDatabase.credentialsDao].
+         * @param context The context that is requesting access to the dao.
+         */
+        fun getInstance(context: Context) = AppDatabase.getInstance(context).credentialsDao()
+    }
+
+    /**
+     * Gets all the credentials stored for the given subscription.
+     * @param subscriptionId The id of the subscription to get the credentials for.
+     * @return The [Credential] stored for the given [subscriptionId] or null if none.
+     * @throws SQLException If there's an error while fetching the credential.
+     */
+    @WorkerThread
+    @Query("SELECT * FROM credentials WHERE subscriptionId=:subscriptionId")
+    fun get(subscriptionId: Long): Credential?
```

Comment:
I suggest the name `getBySubscription`, otherwise the same signature could be interpreted as `get[ById](id: Long)`

Reply:
f3dbfa9f6fa1329031cc58cbf62abf9daf78ba4d